File: 图文混排/图文混排/ViewController.m
//
//  ViewController.m
//  图文混排
//
//  Created by 高级iOS开发 on 2017/10/22.
//  Copyright © 2017年 高级iOS开发. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UILabel *label;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // attachMent - 附件
    NSTextAttachment * attachment = [[NSTextAttachment alloc]init];
    
    attachment.image = [UIImage imageNamed:@"compose_emoticonbutton_background_highlighted"];
    
    // 提示lineheight 大致和字体的大小相等
    CGFloat height = self.label.font.lineHeight;
    attachment.bounds = CGRectMake(0, -4, height, height);
    
    // 图像字符串
    NSAttributedString *imageStr = [NSAttributedString attributedStringWithAttachment:attachment];
    
    // 3定义一个可变的属性字符串
    NSMutableAttributedString * attrstrM = [[NSMutableAttributedString alloc]initWithString:@"我jkhjkhhjg"];
    
    // 修改富文本中的不同文字的样式
    [attrstrM addAttribute:NSForegroundColorAttributeName value:[UIColor blueColor] range:NSMakeRange(0, 5)];
    [attrstrM addAttribute:NSFontAttributeName value:[UIFont systemFontOfSize:20] range:NSMakeRange(0, 5)];
    
   
  
    [attrstrM appendAttributedString:imageStr];
    
    NSAttributedString * text = [[NSAttributedString alloc]initWithString:@"123"];
    
    [attrstrM appendAttributedString:text];
    // 设置属性文本
    self.label.attributedText = attrstrM;
}


@end

File: swift图文混排/swift图文混排/ViewController.swift
//
//  ViewController.swift
//  swift图文混排
//
//  Created by 高级iOS开发 on 2017/10/22.
//  Copyright © 2017年 高级iOS开发. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var label: UILabel!
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // 1 图片附件
        let attachment = NSTextAttachment()
        
        attachment.image = UIImage(named: "compose_emoticonbutton_background_highlighted")
        
        let height = label.font.lineHeight
        attachment.bounds = CGRect(x: 0, y: 0, width: height, height: height)
        
        // 1属性文本
        let imageStr = NSAttributedString(attachment: attachment)
        
        // 2可变的图文字符串
        let attrStrM = NSMutableAttributedString(string: "我")
        attrStrM.append(imageStr)
        let aaaa = NSAttributedString(string: "哈哈")
        attrStrM.append(aaaa)
        
        //3 设置label
        label.attributedText = attrStrM
    }


}

